mesonbuild/interpreter/interpreter.py

```diff
@@ -684,7 +684,7 @@ def func_declare_dependency(self, node, args, kwargs):
             else:
                 if not self.is_subproject() and srcdir / self.subproject_dir in p.parents:
                     continue
-                if p.is_absolute() and p.is_dir() and srcdir / self.root_subdir in [p] + list(p.resolve().parents):
+                if p.is_absolute() and p.is_dir() and srcdir / self.root_subdir in [p] + list(Path(os.path.abspath(p)).parents):
```

Comment:
Sigh, once again pathlib's concrete Path sucks and os.path is infinitely superior. :)